sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/transfer/TransferIdOrError.java

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.storage.blob.transfer;
+
+import androidx.annotation.NonNull;
+
+/**
+ * Package private.
+ *
+ * The type used by the methods in {@link TransferClient} (e.g. upload, download, resume) to
+ * communicate data from background {@link SerialExecutor} to {@link TransferIdMappedToTransferInfo}.
+ *
+ * The data includes the operation type, the actual transfer id of a transfer or an error indicating
+ * the failure detected by TransferClient. Note that this type is used only to channel the error that
+ * TransferClient methods encountered or could detect, this type is not used to channel the real time
+ * error happened in the background worker.
+ */
+public final class TransferIdOrError {
```

Comment:
I feel like maybe we should just make a generic `Result<TSuccess, TErr>` type we can use instead of this. I discovered that this is somehow not a thing that Java includes itself.

Reply:
Seeing the changes here later on. Still think we should make this a more generic name, but maybe not an actual `generic` type. Perhaps something like `TransferOperationResult`?

Reply:
yes, I was thinking the same, generic type is something we can think if we encounter more scenarios where we need to pass value-error through LiveData. `TransferOperationResult` looks good to me.